sam/dns.tf

```diff
@@ -0,0 +1,21 @@
+data "google_dns_managed_zone" "dns_zone" {
+  count = var.enable && contains(["default", "preview"], var.env_type) ? 1 : 0
+
+  provider = google.dns
+  name     = var.dns_zone_name
+}
+
+locals {
+  fqdn = var.enable && contains(["default", "preview"], var.env_type) ? "${local.hostname}${local.subdomain_name}.${data.google_dns_managed_zone.dns_zone[0].dns_name}" : null
```

Comment:
It's getting to be pretty tough to understand what the end result of the fqdn is at this point. I wonder if there's a more readable way we could express this.

Reply:
Nothing obvious comes to mind but I can play around with it